In [4]:
# Import libraries
%pip install statsmodels
import pandas as pd
import statsmodels.api as sm
from scipy import stats
from scipy.stats import chi2_contingency


[notice] A new release of pip is available: 24.3.1 -> 25.0.1
[notice] To update, run: C:\Users\omweg\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\python.exe -m pip install --upgrade pip


   ---------------------------------------- 0.0/9.9 MB ? eta -:--:--
   ---------------------------------------- 0.0/9.9 MB ? eta -:--:--
   - -------------------------------------- 0.3/9.9 MB ? eta -:--:--
   - -------------------------------------- 0.3/9.9 MB ? eta -:--:--
   -- ------------------------------------- 0.5/9.9 MB 540.5 kB/s eta 0:00:18
   ---- ----------------------------------- 1.0/9.9 MB 1.0 MB/s eta 0:00:09
   ------ --------------------------------- 1.6/9.9 MB 1.4 MB/s eta 0:00:07
   ------- -------------------------------- 1.8/9.9 MB 1.4 MB/s eta 0:00:06
   -------- ------------------------------- 2.1/9.9 MB 1.4 MB/s eta 0:00:06
   -------- ------------------------------- 2.1/9.9 MB 1.4 MB/s eta 0:00:06
   ----------- ---------------------------- 2.9/9.9 MB 1.4 MB/s eta 0:00:05
   ------------ --------------------------- 3.1/9.9 MB 1.5 MB/s eta 0:00:05
   -------------- ------------------------- 3.7/9.9 MB 1.5 MB/s eta 0:00:05
   --------------- -------------------

Load Data set

In [2]:
data=pd.read_csv(r"C:\Users\omweg\Desktop\Data Analysis\Real Estate Data Prediction.csv")
data

No  transaction date  house age  distance to the nearest MRT station  \
0      1          2012.917       32.0                             84.87882   
1      2          2012.917       19.5                            306.59470   
2      3          2013.583       13.3                            561.98450   
3      4          2013.500       13.3                            561.98450   
4      5          2012.833        5.0                            390.56840   
..   ...               ...        ...                                  ...   
409  410          2013.000       13.7                           4082.01500   
410  411          2012.667        5.6                             90.45606   
411  412          2013.250       18.8                            390.96960   
412  413          2013.000        8.1                            104.81010   
413  414          2013.500        6.5                             90.45606   

     number of convenience stores  latitude  longitude  \
0                              10  24.98298  121.54024   
1                               9  24.98034  121.53951   
2                               5  24.98746  121.54391   
3                               5  24.98746  121.54391   
4                               5  24.97937  121.54245   
..                            ...       ...        ...   
409                             0  24.94155  121.50381   
410                             9  24.97433  121.54310   
411                             7  24.97923  121.53986   
412                             5  24.96674  121.54067   
413                             9  24.97433  121.54310   

     house price of unit area  
0                        37.9  
1                        42.2  
2                        47.3  
3                        54.8  
4                        43.1  
..                        ...  
409                      15.4  
410                      50.0  
411                      40.6  
412                      52.5  
413                      63.9  

[414 rows x 8 columns]

In [5]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 414 entries, 0 to 413
Data columns (total 8 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   No                                   414 non-null    int64  
 1   transaction date                     414 non-null    float64
 2   house age                            414 non-null    float64
 3   distance to the nearest MRT station  414 non-null    float64
 4   number of convenience stores         414 non-null    int64  
 5   latitude                             414 non-null    float64
 6   longitude                            414 non-null    float64
 7   house price of unit area             414 non-null    float64
dtypes: float64(6), int64(2)
memory usage: 26.0 KB


In [ ]:
# Data set Summary
data.describe()

No  transaction date   house age  \
count  414.000000        414.000000  414.000000   
mean   207.500000       2013.148971   17.712560   
std    119.655756          0.281967   11.392485   
min      1.000000       2012.667000    0.000000   
25%    104.250000       2012.917000    9.025000   
50%    207.500000       2013.167000   16.100000   
75%    310.750000       2013.417000   28.150000   
max    414.000000       2013.583000   43.800000   

       distance to the nearest MRT station  number of convenience stores  \
count                           414.000000                    414.000000   
mean                           1083.885689                      4.094203   
std                            1262.109595                      2.945562   
min                              23.382840                      0.000000   
25%                             289.324800                      1.000000   
50%                             492.231300                      4.000000   
75%                            1454.279000                      6.000000   
max                            6488.021000                     10.000000   

         latitude   longitude  house price of unit area  
count  414.000000  414.000000                414.000000  
mean    24.969030  121.533361                 37.980193  
std      0.012410    0.015347                 13.606488  
min     24.932070  121.473530                  7.600000  
25%     24.963000  121.528085                 27.700000  
50%     24.971100  121.538630                 38.450000  
75%     24.977455  121.543305                 46.600000  
max     25.014590  121.566270                117.500000

# Multiple Line regression
y = B0 + B1 * x1 + B2 * x2 + B3 * x3 + E

Where :
- y is the dependent variable (the variable we are trying to predict) = House Price
- B0 is the intercept or constant term
- B1, B2, B3 are the coefficients of the independent variables
- E is the error term (the difference between the observed and predicted values)
- x1 = House Age
- x2 = Distance of MRT Satation
- x3 = No of convience stores


In [8]:
X = data[["house age", "distance to the nearest MRT station", "number of convenience stores"]]
X = sm.add_constant(X)  # Add constant term for regression
y = data["house price of unit area"]

# Multiple Linear Regression
model = sm.OLS(y, X).fit()
print("\nMultiple Linear Regression Results: Predicting Average Score")
print(model.summary())


Multiple Linear Regression Results: Predicting Average Score
                               OLS Regression Results                               
Dep. Variable:     house price of unit area   R-squared:                       0.541
Model:                                  OLS   Adj. R-squared:                  0.538
Method:                       Least Squares   F-statistic:                     161.1
Date:                      Tue, 04 Mar 2025   Prob (F-statistic):           5.44e-69
Time:                              10:23:05   Log-Likelihood:                -1506.5
No. Observations:                       414   AIC:                             3021.
Df Residuals:                           410   BIC:                             3037.
Df Model:                                 3                                         
Covariance Type:                  nonrobust                                         
                                          coef    std err          t      P>|t|      [0.

# Overall Analysis
* **Adjusted R-squared(0.538)** - states that 53.8% of the variance of the house prices depend on the independent variables(house age, distance to the nearest MRT station, number of convenience stores)
* **Goodness of fit(5.44e-69)** - states that the it is significantly lower than the p - value of 0.05 hence it is staistically significant here by indicating that at least one of the independant variables can be used to calculate the price of the house.
* **Regression equation:**  y = 42.9773 + -0.2529(x1) + -0.0054(x2) + 1.2974(x3) + 0

    * **const(42.9773)** - this shows the value o the house if all other variables are zero
    * **house age (-0.2529)** - this shows that the age  of the house is negatively correlated with the price of the house and statistically significant. This means that the older the house, the lower the price. Therefore decreasing by -25.29% per year.
    * **distance to the nearest MRT station(-0.0054)** - this shows that the distance to the nearest MRT station is negatively correlated with the price of the house and statistically significant. This means that the further away the house is from the nearest MRT station, the lower the price. A decrease by -0.54% per unit of distance.
    * **number of convenience (1.2974)** - this shows that the number of convience stores positively correlated with the price of the house and statistically significant. This means that the more convenience stores there are near the house the higher the price of the house. An increase of 1.30 times more per convenience store.
    * **E** (0) - Assume there was no other external factor




